### PR TITLE
[TBE-116] try a new way to check if we need to create a heroku app

### PR DESCRIPTION
**Pull request by jnf**
## Link to pivotal/JIRA issue
[TBE-116](https://codeforamerica.atlassian.net/browse/TBE-116)
## Is PM acceptance required? (delete one)
No - merge after code review approval

**Reminder**: merge main into this branch and get green tests before merging to main

## What was done?
- Refactors the heroku build script into 2 separate scripts: one for building, one for destroying.
- `opened`, `reopened`, and `synchronize` action verbs all trigger the building script
- `closed` action verb triggers the destroying script
- we don't actually try to build the environment every time; instead we ask heroku if an environment exists, and, if it doesn't, we build it.

## How to test?
Only 2 ways I've come up with to test this:
- ~Merge it to main and hope~
- ~Merge (or cherry-pick?) this branch into another, unrelated branch and then open a PR to test the build processes. If we're gonna do that I may take a few moments to clean up the commits first so it's less mess in someone else's work.~

### Update
I closed an earlier version of this PR and the [new 'closed' action did the thing](https://github.com/codeforamerica/vita-min/actions/runs/11468395978/job/31913413536)! Turns out, and this feels very obvious in retrospect, the actions that run are the actions defined on the branch.

And, opening this version of the PR triggered the [new build script](https://github.com/codeforamerica/vita-min/actions/runs/11469188183/job/31915909902?pr=4900)!

### Risk Assessment
We absolutely rely on heroku for acceptance testing. Any changes to how he provision and tear down environments could impact that larger software delivery flow. We should keep a close eye on heroku builds for awhile even after this is tested and merged.

[TBE-116]: https://codeforamerica.atlassian.net/browse/TBE-116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/heroku-remove-app.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Heroku Pull Request
```
Comment (embarnard) [inline .github/workflows/heroku-remove-app.yml:1]: Excuse my lack of gitflow knowledge, where do these workflows get called? is that just setup in git? Or does it just know to run this workflow whenever there is a "closed" git action?

Comment (jnf) [reply .github/workflows/heroku-remove-app.yml:1]: they're executed but github automatically. github looks in the `.github/workflows` directory for scripts to register/run. so yeah, your intuition is correct -- when the 'closed' action happens, github checks for a workflow triggered by that event.

File: .github/workflows/heroku-remove-app.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          heroku_email: jheath@codeforamerica.org
+          heroku_app_name: ${{ env.HEROKU_APP_NAME }}
+          justlogin: true
+
+      - name: Predestroy the app associated with the closed PR
+        run: heroku run rails heroku:review_app_predestroy --app=${{ env.HEROKU_APP_NAME }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: For realsies destroy the Heroku app associated with the closed PR
```
Comment (embarnard) [inline .github/workflows/heroku-remove-app.yml:26]:
```suggestion
      - name: For releases destroy the Heroku app associated with the closed PR
```

**Review by embarnard — APPROVED** [review]
Good work! I think this all makes sense, just that one comment

**Pull request merged**